packages/zudoku/src/vite/plugin-metadata.ts

```diff
@@ -18,11 +19,14 @@ const viteBuildMetadata = (): Plugin => {
           }
         }
       }
-
-      writeFileSync(
-        path.join(__dirname, "graph.json"),
-        JSON.stringify(deps, null, 2),
-      );
+      try {
+        await writeFile(
+          path.join(__dirname, "graph.json"),
+          JSON.stringify(deps, null, 2),
+        );
+      } catch (error) {
+        logger.error("Error writing graph.json:", error);
```

Comment:
We'd need to re-throw the error, otherwise the build will not fail, but it should.